### ISSUE TITLE
Not able to reproduce results on OULU

### ISSUE THREAD

**Issue by shlokk** [ZitongYu/CDCN#27]
Hi, 
I'm not able to reproduce results on OULU dataset. 
Even after 700 epochs , the loss as low as mentioned in other issue. 
epoch:721, Test:  ACC= 0.4100, APCER= 0.7375, BPCER= 0.0000, ACER= 0.3688
epoch:722, mini-batch: 50, lr=0.000050, Absolute_Depth_loss= 0.0279, Contrastive_Depth_loss= 0.0032
epoch:722, mini-batch:100, lr=0.000050, Absolute_Depth_loss= 0.0264, Contrastive_Depth_loss= 0.0030
epoch:722, mini-batch:150, lr=0.000050, Absolute_Depth_loss= 0.0278, Contrastive_Depth_loss= 0.0031
epoch:722, Train:  Absolute_Depth_loss= 0.0271, Contrastive_Depth_loss= 0.0031

Differences from original setup:
1) I have 20 random frames while training, instead of all the frames. The reason I only have 20 frames is that it is very expensive to use PRNET to generate depth, so I have only used 20. 
2) For bbox you have '.dat' file, I did;t find that in the dataset and neither in PRNet output. So, I have used the BBOX from the dataset. They look decent and cover the face mostly.
3) Hyperparameters: I have used default hyperparameters: i.e bs 7 , lr 0.0001 and --gamma 0.05.
Let me know what you think can be the most pressing issue and why I'm not able to reprdouce the results.

**Comment by 22wei22** [comment]
您好，请问您方便分享一下得到人脸深度图的代码么？我使用的这个项目，不过我的一些图片检测不到人脸https://github.com/clks-wzz/PRNet-Depth-Generation 1334637558@qq.com

**Comment by 610821216** [comment]
> Hi,
> I'm not able to reproduce results on OULU dataset.
> Even after 700 epochs , the loss as low as mentioned in other issue.
> epoch:721, Test: ACC= 0.4100, APCER= 0.7375, BPCER= 0.0000, ACER= 0.3688
> epoch:722, mini-batch: 50, lr=0.000050, Absolute_Depth_loss= 0.0279, Contrastive_Depth_loss= 0.0032
> epoch:722, mini-batch:100, lr=0.000050, Absolute_Depth_loss= 0.0264, Contrastive_Depth_loss= 0.0030
> epoch:722, mini-batch:150, lr=0.000050, Absolute_Depth_loss= 0.0278, Contrastive_Depth_loss= 0.0031
> epoch:722, Train: Absolute_Depth_loss= 0.0271, Contrastive_Depth_loss= 0.0031
> 
> Differences from original setup:
> 
> 1. I have 20 random frames while training, instead of all the frames. The reason I only have 20 frames is that it is very expensive to use PRNET to generate depth, so I have only used 20.
> 2. For bbox you have '.dat' file, I did;t find that in the dataset and neither in PRNet output. So, I have used the BBOX from the dataset. They look decent and cover the face mostly.
> 3. Hyperparameters: I have used default hyperparameters: i.e bs 7 , lr 0.0001 and --gamma 0.05.
>    Let me know what you think can be the most pressing issue and why I'm not able to reprdouce the results.

hello  i dont know how to use bbox generate '.dat' file.....can u teach me or share me file....plz

mail: 610821216@gms.ndhu.edu.tw

**Comment by liuyuchen6666** [comment]
@shlokk Hello, thans a lot. I want to request for the data preparation. Specifically, how can I separate the video into frames?

**Comment by jamesdongdong** [comment]
> @shlokk Hello, thans a lot. I want to request for the data preparation. Specifically, how can I separate the video into frames?

Sorry to disturb you. Do you already know how to do data preparation?
